contracts/utils/math/Math.sol

```diff
@@ -163,8 +162,7 @@ library Math {
             // Factor powers of two out of denominator and compute largest power of two divisor of denominator. Always >= 1.
             // See https://cs.stackexchange.com/q/138556/92363.
 
-            // Does not overflow because the denominator cannot be zero at this stage in the function.
-            uint256 twos = denominator & (~denominator + 1);
+            uint256 twos = denominator & (0 - denominator);
```

Comment:
That is indeed a good solution !

Reply:
For the record, before solidity v8, the code used to be [`uint256 twos = -denominator & denominator;`](https://github.com/Uniswap/v3-core/blob/412d9b236a1e75a98568d49b1aeb21e3a1430544/contracts/libraries/FullMath.sol#L64) but solidity 0.8 forbid the `-x` syntax for unsigned integers. Doing `(0-x)` indeed fixes it.